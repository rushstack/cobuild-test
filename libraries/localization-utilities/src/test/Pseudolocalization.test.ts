// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { getPseudolocalizer } from '../Pseudolocalization';
import type { IPseudolocaleOptions } from '../interfaces';

describe(getPseudolocalizer.name, () => {
  it('gets distinct pseudolocalizers', () => {
    const input: 'text' = 'text';

    const fooOptions: IPseudolocaleOptions = {
      prepend: '-Foo-',
      append: '-Foo-'
    };
    const fooLocale: (str: string) => string = getPseudolocalizer(fooOptions);
    const foo1: string = fooLocale(input);

    const barOptions: IPseudolocaleOptions = {
      prepend: '-Bar-',
      append: '-Bar-'
    };

    const barLocale: (str: string) => string = getPseudolocalizer(barOptions);

    const bar1: string = barLocale(input);
    const foo2: string = fooLocale(input);
    const bar2: string = barLocale(input);

    expect(foo1).toEqual(foo2);
    expect(bar1).toEqual(bar2);

    expect(foo1).toMatchSnapshot('foo');
    expect(bar1).toMatchSnapshot('bar');
  });
});
