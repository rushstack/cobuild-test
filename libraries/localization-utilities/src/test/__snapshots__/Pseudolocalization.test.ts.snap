// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPseudolocalizer gets distinct pseudolocalizers: bar 1`] = `"-Bar-ţēxţ-Bar-"`;

exports[`getPseudolocalizer gets distinct pseudolocalizers: foo 1`] = `"-Foo-ţēxţ-Foo-"`;
